File: core/src/com/broll/gainea/server/core/cards/impl/C_TeleportLake.java
package com.broll.gainea.server.core.cards.impl;

import com.broll.gainea.server.core.cards.AbstractTeleportCard;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.Location;
import com.broll.gainea.server.core.utils.LocationUtils;

import java.util.List;
import java.util.stream.Collectors;

public class C_TeleportLake extends AbstractTeleportCard {
    public C_TeleportLake() {
        super(48, "Küstenreise", "Bewegt eine Truppe zu einem beliebigem Meer auf der gleichen Karte");
    }

    @Override
    public List<Location> getTeleportTargets(Location from) {
        return LocationUtils.filterByType(from.getContainer().getExpansion().getAllLocations(), AreaType.LAKE).collect(Collectors.toList());
    }
}

File: server/src/main/com/broll/gainea/server/core/cards/impl/play/C_TeleportLake.kt
package com.broll.gainea.server.core.cards.impl.play

import com.broll.gainea.server.core.Game
import com.broll.gainea.server.core.cards.TeleportCard
import com.broll.gainea.server.core.map.AreaType.LAKE
import com.broll.gainea.server.core.map.Location

class C_TeleportLake : TeleportCard(47, "Küstenreise", "Bewegt eine Truppe zu einem beliebigem Meer auf der gleichen Karte") {

    override fun getTeleportTargets(from: Location) = from.container.expansion.allAreas.filter { it.type == LAKE }

    override fun validFor(game: Game) = game.map.allAreas.any { it.type == LAKE }
}
